File: C/Seq/sgemm.c
void sgemm(int m, int n, int k, float *a, float *b, float *c)
{
  for (int i = 0; i < m; i++)
  {
    for (int j = 0; j < n; j++)
    {
      for (int l = 0; l < k; l++)
      {
        c[i * n + j] += a[i * k + l] * b[l * n + j];
      }
    }
  } 
}

File: Python/Seq/sgemm.py
# SGEMM using numpy arrays
import numpy as np

N = 1000
M = 1000
K = 1000

A = np.random.rand(N, M)
B = np.random.rand(M, K)
C = np.random.rand(N, K)

for i in range(N):
    for j in range(K):
        for k in range(M):
            C[i][j] += A[i][k] * B[k][j]
